src/SuperHero.java

```diff
@@ -6,20 +6,26 @@ public class SuperHero {
 
 
     public void attack(SuperHero opponent) {
-        System.out.println("SuperHero  " + name + "  attacks");
+        System.out.println("SuperHero   " + name + "   attacks ");
         opponent.takeDamage(attackPower);
     }
 
     public void takeDamage(int damage) {
-        System.out.println("SuperHero  " + name + "  receives" + damage + "  damage");
         health -= damage;
+        if (health < 0 ) {
+            health = 0;
+            System.out.println("SuperHero  " + name + " took " + damage + "damages. Current health:  " + health);
+        }
+        if (health == 0){
+        System.out.println("SuperHero  " + name + " died in an epic battle");
```

Comment:
I believe this can be simplified to only one `if`

```java
...
health -= damage;

if (health <= 0) {
     health = 0; // this already means hero is dead
     System.out.println("SuperHero  " + name + " died in an epic battle");
}
```